FinSearchUnified/Bundle/SearchBundleFindologic/QueryBuilder.php

```diff
@@ -166,4 +166,105 @@ private function getClientIp()
 
         return $ipAddress;
     }
+
+    /**
+     * @return array
+     */
+    public function getParameters()
+    {
+        return $this->parameters;
+    }
+
+    /**
+     * @param string $query
+     */
+    public function addQuery($query)
+    {
+        $this->parameters['query'] = urldecode($query);
+    }
+
+    /**
+     * @param float $min
+     * @param float $max
+     */
+    public function addPrice($min, $max)
+    {
+        $this->addParameter('price', ['min' => urldecode($min), 'max' => urldecode($max)]);
+    }
+
+    /**
+     * @param string $key
+     * @param string $value
+     */
+    public function addFilter($key, $value)
+    {
+        if (!empty($this->getParameter($key))) {
+            $this->addParameter($key, urldecode($value));
+        } else {
+            $this->addParameter($key, [urldecode($value)]);
+        }
+    }
+
+    /**
+     * @param array $categories
+     */
+    public function addCategories(array $categories)
+    {
+        $this->addParameter('cat', $categories);
+    }
+
+    /**
+     * @param string $order
+     */
+    public function addOrder($order)
+    {
+        $this->parameters['order'] = $order;
+    }
+
+    /**
+     * @param int $firstResult
+     */
+    public function setFirstResult($firstResult)
+    {
+        $this->parameters['first'] = $firstResult;
+    }
+
+    /**
+     * @param int $maxResults
+     */
+    public function setMaxResults($maxResults)
+    {
+        $this->parameters['count'] = $maxResults;
+    }
+
+    /**
+     * @param string $key
+     *
+     * @return mixed|null
+     */
+    private function getParameter($key)
+    {
+        if (!isset($this->parameters['attrib']) || !isset($this->parameters['attrib'][$key])) {
```

Comment:
Since the string `attrib` is used quite often:
* Extract it into a constant (can be a member of QueryBuilder), e.g. `QueryBuilder::PARAMETER_KEY_ATTRIB`.
* Make sure to use the constant only here, not in the tests.